### PR TITLE
Adding Resource Group As Optional Param

### LINKED ISSUE #256: Specify resource group or app service environment.
Deployment failed when there are multiple app services which have same name. Of course we can't create Non-ASE app service which have duplicate name because all app services have same domaim suffix (azurewebsites.net) . But ASE app service can have duplicate name because ASE has unique domain suffix (asename.appserviceenvironment.net) each other. 

This action seems to search app service instance by the name only, but the name may not be unique in ASE app service. So I think this action should have option to specify the resource group or environment for uniqueness.

workflow file is below, 
```yaml
      - name: Deploy to Azure Web App in ILB ASE v2
        id: deploy-to-app2
        uses: azure/webapps-deploy@v2
        with:
          app-name: 'app2'
          package: .
```

but there are multiple app service named "app2".
image

and workflow fail with this error.
```
Run azure/webapps-deploy@v2
  with:
    app-name: app2
    package: .
    slot-name: production
  env:
    AZURE_HTTP_USER_AGENT: 
    AZUREPS_HOST_ENVIRONMENT: 
Error: Deployment Failed with Error: Error: Multiple resource group found for resource nameapp2
```

### PR DESCRIPTION
**Pull request by shpraka**
This helps fix #256

### CODE REVIEW

**Review by kumaraksh1 — APPROVED** [review]
LGTM!!

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "@actions/core": "^1.10.0",
     "@actions/github": "^4.0.0",
     "actions-secret-parser": "^1.0.4",
-    "azure-actions-appservice-rest": "^1.3.3",
+    "azure-actions-appservice-rest": "^1.3.6",
```
Comment (surenderssm) [inline package.json:37]: Is this upgrade intentional?

File: action.yml [hunk 2]
```diff
@@ -25,6 +25,10 @@ inputs:
   startup-command:
     description: 'Enter the start up command. For ex. dotnet run or dotnet run'
     required: false
+  resource-group-name:
+    description: 'Enter the resource group name of the web app'
+    required: false
+
```
Comment (surenderssm) [inline action.yml:31]: NIT : new line?

File: package.json (hunk 1 above)
Comment (shpraka) [reply package.json:37]: yes, we published the new version on our dependent repo.

File: action.yml (hunk 2 above)
Comment (shpraka) [reply action.yml:31]: Will resolve this comment in next PR.

**Pull request merged**